### ISSUE TITLE
Reverting Register calls

### ISSUE THREAD

**Issue by ggwpez** [hyperledger-labs/go-perun#7]
**Location**&emsp; `[client]`, `[channel]`, `[backend/eth/channel]`.

**Discussed in**&emsp; `DST/dev` chat.

**Context**&emsp; When both users try to close a *finalized* channel at the same time, one of their transactions can revert with `channel already concluded`.  
This results from both parties sending the `conclude` transaction Thanks to @manoranjith for finding this.  
`channel.Settle` has a `secondary` flag which can be used to prevent this from happening.  

A makeshift solution is to let one party wait two block duration.

**Changes**&emsp; 
There are at least two possibilities here:
1. Add a `Secondary` flag to `channel.Register` which is only used for *final* states.
2. Add some kind of `wait` function that would be called before `Register`.  

I think we should keep `settle` and `register` consistent, so either use two `wait` functions or two `secondary` flags.

**Acceptance criteria**&emsp; The user can prevent *go-perun* from sending a reverting register transaction.

**Testing**&emsp; All of our Two-party tests use a simulated ETH-chain which makes this unlikely to trigger.  
It is still be worth trying it out though. 

**Implementation hints**&emsp; A sleep can be added [here](https://github.com/hyperledger-labs/go-perun/blob/35786ace33d74e1c40a1073e7f329a025632fd24/backend/ethereum/channel/conclude.go#L55) to make it more likely to fail.

**Comment by manoranjith** [comment]
The second possibility seems more interesting to me compared to the first one.

Because in the first case, the secondary flag has to be received at the top level API and should be propagated across different methods in the core of go-perun as well as the backend.

Whereas, in the second case, the user of the go-perun calls the `wait` function before `Register`/`Settle` as necessary. Also the secondary logic need to be included in any part of the go-perun (both core & backend). This also means the same objective achieved with lesser number of lines of code.

**Comment by ggwpez** [comment]
> The second possibility seems more interesting to me compared to the first one.

Im not sure if a `waitNBlocks` function instead of a `secondary` flag makes it less block chain agnostic.   
In an instant-finality (but high TX fee) chain for example, this would complicate things.
What do you think @matthiasgeihs ?

**Comment by matthiasgeihs** [comment]
I don't see a big problem with having a transaction revert. If the transaction reverts because the channel has already been registered, that's ok, because the goal was to register the channel. The framework should just acknowledge that the channel has been registered, log the transaction error, but do not return the error to the user, as the goal of registering the channel has been fulfilled.

**Comment by ggwpez** [comment]
> I don't see a big problem with having a transaction revert.  

Well it is not directly a "problem" because everything works as expected.  
It is more of an optimization. We tried to build *go-perun* to be as gas efficient as possible.  
This also means to prevent reverting transactions, hence why we introduced the `secondary` flag on the `Settle` call.  

It is sure up for discussion on how far we should go here.
